build.sbt

```diff
@@ -685,7 +688,7 @@ lazy val compilerSettings = Seq(
 
     val base = {
       if (scalaBinaryVersion.value == "3") {
-        minimal :+ "-deprecation"
+        minimal :+ "-Wconf:cat=deprecation:s"
```

Comment:
Curious why this was needed?

Reply:
```
[error] -- Error: /home/gregg/lucid/enumeratum/macros/src/main/scala/enumeratum/compat/EnumMacros.scala:109:32 
[error] 109 |        val enclosingModule = c.enclosingClass match {
[error]     |                              ^^^^^^^^^^^^^^^^
[error]     |method enclosingClass in trait Enclosures is deprecated since 2.11.0: c.enclosingTree-style APIs are now deprecated; consult the scaladoc for more information
```

and a few more identical errors